roles/sft-server/vars/main.yml

```diff
@@ -0,0 +1,30 @@
+# NOTE: considered to be constants due to high precedence
+
+SFT_ARTIFACT_TARGET_PATH: '/tmp/sft-artifact.tar.gz'
+SFT_BIN_PATH: '/usr/local/bin/sftd'
+
+# NOTE: currently hardcoded in source, but will soon be configurable
+SFTD_SIGNALING_PORT: 8585
+SFTD_METRICS_PORT: 49090
+
+SFTD_USER: 'sft'
+SFTD_GROUP: 'sft'
+
+
+
+NGINX_CONF_PATH: '/etc/nginx/nginx.conf'
+NGINX_CONF_DIR: '/etc/nginx/conf.d'
+NGINX_DHPARAMS_PATH: '/etc/ssl/nginx-dhparams.pem'
+
+
+CERTBOT_CONF_PATH: '/etc/letsencrypt'
+CERTBOT_WEBROOT_PATH : '/usr/share/nginx/certbot'
+
+# NOTE: the following two paths cannot really be configured for a certbot invocation
+CERTBOT_LIVE_PATH: "{{ CERTBOT_CONF_PATH }}/live"
+CERTBOT_ARCHIVE_PATH: "{{ CERTBOT_CONF_PATH }}/archive"
+
+CERTBOT_CERT: "{{ CERTBOT_LIVE_PATH }}/{{ sft_fqdn }}/cert.pem"
+CERTBOT_PRIVKEY: "{{ CERTBOT_LIVE_PATH }}/{{ sft_fqdn }}/privkey.pem"
+CERTBOT_CHAIN: "{{ CERTBOT_LIVE_PATH }}/{{ sft_fqdn }}/chain.pem"
+CERTBOT_FULLCHAIN: "{{ CERTBOT_LIVE_PATH }}/{{ sft_fqdn }}/fullchain.pem"
```

Comment:
Setting these variables is not needed; you can omit all of the CERTBOT_ variables here, those are the defaults anyway. (both here and in the `certbot ... ` invocation.)

Reply:
In this case, it's partially not true, because they are also being used in the `stfd`'s vhost (nginx) configuration. My general approach is, if I use a value (with the same context/purpose) more than once, I create a variable/constant.

About the default values: I often tent to explicitly set values even if they are the default - for the sake of documentation. Also,  when searching through files explicit default configuration is easier to spot, compared to non-written defaults. Sure, there is official documentation, but in my experience at first people search in the code.

Would you be okay with these approaches, or do you have other thoughts on this?

Reply:
IMHO, adding variables suggests configurability, or the need for it.

I'd feel much more confident with hardcoding stuff (or just letting the defaults take over) in cases where we don't explicitly want to devitate from the defaults, as this makes the code more readable.

If the location of these certificates needs to be documented, I'd rather see this documented in comments than in variables - or we end up asking ourselves if there's any instances where we /do/ configure it to a different location.

Reply:
It seems the only place the location of certificates needs to be set, is inside the vhost file. It adheres to the defaults from certbot, and they are not needed elsewhere, so two lines like this: https://github.com/wireapp/ansible-restund/blob/master/templates/nginx-stream.conf.j2#L16-L17 and nothing in this file, and nothing for the certbot invocation itself also does the trick.
I find this PR adds a lot of noise of stuff that is no different than the default configuration. 50 lines need to be mentally scanned to figure out if and where one line might be non-default and relevant to look at.
I'm with Florian here and prefer fewer lines of code where possible.

Reply:
Worse, we don't configure `certbot.ini.j2` with the variables exposed here.

So while one would assume setting this variable changes the path certbot certificates are stored, it only affects the location nginx tries to read them from - and configuring this to anything else than the default will fail.

Reply:
> Worse, we don't configure `certbot.ini.j2` with the variables exposed here.

I would have loved to do this, but certbot behaves somewhat weird when trying to configure it via file(s).

In any case, I removed the four constants pointed out above.